tools/mkcomposefs.c

```diff
@@ -129,6 +129,7 @@ static char *unescape_string(const char *escaped, size_t escaped_size,
 		if (c == '\\') {
 			if (escaped >= escaped_end) {
 				*err = make_error("No character after escape");
+				free(res);
```

Comment:
Let's use `cleanup_free` instead, it's designed for this.  The return path will need to `steal_pointer`.